build.bat

```diff
@@ -1,31 +1,52 @@
 @echo off
 rem Rime build script for msvc toolchain.
-rem 2014-12-30  Chen Gong <chen.sst@gmail.com>
+rem Maintainer: Chen Gong <chen.sst@gmail.com>
 
 setlocal
 set BACK=%CD%
 
 if exist env.bat call env.bat
 
 set OLD_PATH=%PATH%
-if defined DEV_PATH set PATH=%OLD_PATH%;%DEV_PATH%
+if defined DEVTOOLS_PATH set PATH=%OLD_PATH%;%DEVTOOLS_PATH%
 path
 echo.
 
 if not defined RIME_ROOT set RIME_ROOT=%CD%
 echo RIME_ROOT=%RIME_ROOT%
 echo.
 
+if defined BOOST_ROOT (
+  if exist "%BOOST_ROOT%\boost" goto boost_found
+)
+echo Error: Boost not found! Please set BOOST_ROOT in env.bat.
+exit /b 1
+:boost_found
 echo BOOST_ROOT=%BOOST_ROOT%
 echo.
 
-if defined CMAKE_INSTALL_PATH set PATH=%PATH%;%CMAKE_INSTALL_PATH%
+if not defined BJAM_TOOLSET (
+  set BJAM_TOOLSET=msvc-%VisualStudioVersion%
+)
+
+if not defined CMAKE_GENERATOR (
+  set CMAKE_GENERATOR="Visual Studio 14 2015"
+)
 
-set CMAKE_GENERATOR="Visual Studio 14 2015"
-set CMAKE_TOOLSET="v140_xp"
+if not defined PLATFORM_TOOLSET (
+  set PLATFORM_TOOLSET=v140_xp
+)
+
+if not defined CMAKE_TOOLSET (
+  set CMAKE_TOOLSET="%PLATFORM_TOOLSET%"

```

Comment:
Question: are the quotes required in `cmake -T"v140_xp"` ?
`PLATFORM_TOOLSET` which is also referenced in `*.vcxproj` files should NOT have quotes.

Reply:
Since it does not include spaces, no.

Quotes are automatically removed in project files.

Reply:
If `CMAKE_TOOSET` is not currently in use anywhere else, we can delete it and use `PLATFORM_TOOLSET` directly.

Reply:
OK, then I'll change to `PLATFORM_TOOLSET`
It is also used in the other PR https://github.com/rime/weasel/pull/112 where "CMAKE_TOOSET" is not appropriate.